### PR TITLE
add moderation sections to code of conduct

### PR DESCRIPTION
**Pull request by michaelficarra**
As discussed in https://github.com/jashkenas/underscore/pull/2454.

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -59,12 +59,28 @@ reported by contacting the project team at jashkenas@gmail.com. The project team
...
 faith may face temporary or permanent repercussions as determined by other
 members of the project's leadership.
 
+## Moderation
+
+Edits of another user's comment must be clearly marked with **edit**,
+the moderator's username, and a timestamp for each occurrence. The only acceptable
+reasons for editing another user's comment are:
+
+1. to edit out violations of this Code of Conduct. These edits must include a rationale.
```
Comment (akre54) [inline CODE_OF_CONDUCT.md]: For clarification does "violations" refer to "behavior that contributes to creating a positive environment" (L17-21 above) or "unacceptable behavior by participants" (L25-32, above)?

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -59,12 +59,28 @@ reported by contacting the project team at jashkenas@gmail.com. The project team
...
+1. to direct future readers to a relevant point later in the conversation (usually the resolution). These edits must be append-only.
```
Comment (jridgewell) [inline CODE_OF_CONDUCT.md]: Please use `2.`

Comment (michaelficarra) [inline CODE_OF_CONDUCT.md]: I typically don't do that for maintainability and VCS reasons, but okay.

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (michaelficarra) [inline CODE_OF_CONDUCT.md]: Lines 25-32 are simply examples of violations of lines 17-21.

Comment (akre54) [inline CODE_OF_CONDUCT.md]: That's not how I read it. Both are listed as "examples" (pro and con). They cover very different ground.

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -59,12 +59,32 @@ reported by contacting the project team at jashkenas@gmail.com. The project team
...
+## Moderation
+
+Edits of another user's comment must be clearly marked with "**edit**", the
+moderator's username, and a timestamp for each occurrence. The only acceptable
+reasons for editing another user's comment are:
+
+1. to edit out violations of this Code of Conduct. These edits must include a
+   rationale.
+2. to direct future readers to a relevant point later in the conversation
+   (usually the resolution). These edits must be append-only.
```
Comment (megawac) [inline CODE_OF_CONDUCT.md]: Can we modify this slightly to permit silent edits for creating or updating cross-references to other issues.

I often edit PRs with things like `fixes #123`

Comment (akre54) [inline CODE_OF_CONDUCT.md]: What about code formatting (like backtick fencing)?

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (michaelficarra) [inline CODE_OF_CONDUCT.md]: Oh sorry, I misspoke. They're both examples of 5-10 ("Our Pledge"); the first set contains examples of acceptable behaviour and the other set contains examples of unacceptable behaviour. The violation here then refers to lines 5-10. I will reword it to be "violations of Our Pledge" if you think that is more clear.

File: CODE_OF_CONDUCT.md (hunk 3 above)
Comment (michaelficarra) [inline CODE_OF_CONDUCT.md]: @megawac This section discusses editing of others' comments. You are free to edit your own PR comment before anyone has responded, or via an append-only change once someone else has responded. You are also free to leave a comment attributed to your own name that says `fixes #123`. I don't think it would be right to silently alter someone's speech, no matter how small you feel it is. Notice that I've not given an exception for correcting someone else's typos/grammar/formatting because that would impose on their freedom of expression (answering @akre54's question).

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (akre54) [inline CODE_OF_CONDUCT.md]: Ok sure. But does that mean comments can be edited if they don't use "welcoming and inclusive language"? Or only if they use "sexualized language or imagery and unwelcome sexual attention or advances"?

Sorry to be a stickler, just trying to work out what this means in practice.

File: CODE_OF_CONDUCT.md (hunk 3 above)
Comment (akre54) [inline CODE_OF_CONDUCT.md]: Cool I agree with this. I feel like in general if you're editing your own comment to do more than minor grammar fixes or rewording you should make a note of it. It becomes really hard to track a discussion when it's not clear what changed when.

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (michaelficarra) [inline CODE_OF_CONDUCT.md]: Quoting the "Our Pledge" section,

>  ... making participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, gender identity and expression, level of experience, nationality, personal appearance, race, religion, or sexual identity and orientation.

So it appears to be no to the former and yes to the latter because it would be harassment.

**Comment by michaelficarra** [comment]
So are we ready to :ship: this?

**Comment by akre54** [comment]
:+1:
